examplescripts/ubuntu_deploy.py

```diff
@@ -1,123 +1,202 @@
+wg_quick = None
+
 
 payment_detail = zos.pools.create(cu=100, su=100, farm="freefarm", currencies=["TFT"])
 print(payment_detail)
 
-if not j.clients.stellar.find("ubuntu_wallet"):
-    j.clients.stellar.create_testnet_funded_wallet("ubuntu_wallet")
+if not j.clients.stellar.find(WALLET_NAME):
+    j.clients.stellar.create_testnet_funded_wallet(WALLET_NAME)
 
-wallet = j.clients.stellar.get("ubuntu_wallet")
+wallet = j.clients.stellar.get(WALLET_NAME)
 
 txs = zos.billing.payout_farmers(wallet, payment_detail)
+print(txs)
 pool = zos.pools.get(payment_detail.reservation_id)
+# wait until pool is activated
 while pool.cus == 0:
     pool = zos.pools.get(payment_detail.reservation_id)
     sleep(1)
 
+print("compute units available:", pool.cus)
+print("storage units available:", pool.sus)
+
+# create ssh key
 if not j.sals.fs.exists("/tmp/.ssh"):
     j.core.executors.run_local('mkdir /tmp/.ssh && ssh-keygen -t rsa -f /tmp/.ssh/id_rsa -q -N "" ')
+j.core.executors.run_local("ssh-add /tmp/.ssh/id_rsa")
 
 ssh_cl = j.clients.sshkey.get("ubuntu_script")
 ssh_cl.private_key_path = "/tmp/.ssh/id_rsa"
 ssh_cl.load_from_file_system()
 ssh_cl.save()
 
-print("compute units available:", pool.cus)
-print("storage units available:", pool.sus)
-
-network_name = "demo_network_script"
-ip_range = "10.110.0.0/16"
-ip_container_1 = "10.110.4.12"
-
-ip_container_2 = "10.110.4.15"
-
-network = zos.network.create(ip_range, network_name)
-nodes = zos.nodes_finder.nodes_by_capacity(pool_id=pool.pool_id)
-access_node = list(filter(zos.nodes_finder.filter_public_ip4, nodes))[0]
-zos.network.add_node(network, access_node.node_id, "10.110.2.0/24", pool.pool_id)
-wg_quick = zos.network.add_access(network, access_node.node_id, "10.110.3.0/24", ipv4=True)
+network = zos.network.load_network(network_name)
+if not network:
+    network = zos.network.create(ip_range, network_name)
+
+network_node_ids = set()
+network_used_ranges = set()
+node_ranges = {}
+
+# get used range in the network
+for resource in network.network_resources:
+    node_ranges[resource.info.node_id] = resource.iprange
+    network_node_ids.add(resource.info.node_id)
+    network_used_ranges.add(resource.iprange)
+    for peer in resource.peers:
+        network_used_ranges.add(peer.iprange)
+
+# search for access node with public IPv4
+pool_nodes = list(zos.nodes_finder.nodes_by_capacity(pool_id=pool.pool_id))
+for node in filter(zos.nodes_finder.filter_public_ip4, pool_nodes):
+    if zos.nodes_finder.filter_is_up(node):
+        access_node = node
+        break
+else:
+    raise Exception("can not find a node with ipv4 that can be used as access node for the cluster")
+
+freerange1 = get_free_iprange(ip_range, network_used_ranges)
+zos.network.add_node(network, access_node.node_id, freerange1, pool.pool_id)
+network_used_ranges.add(freerange1)
+
+freerange2 = get_free_iprange(ip_range, network_used_ranges)
+wg_quick = zos.network.add_access(network, access_node.node_id, freerange2, ipv4=True)
```

Comment:
why are you always adding access to the network? if the network is already created, there is no need to add access.